source/backend/api/Services/DocumentSyncService.cs

```diff
@@ -57,28 +57,40 @@ public ExternalBatchResult SyncMayanMetadataTypes(SyncModel model)
 
             // Add the metadata types not in mayan
             IList<Task<ExternalResult<MetadataType>>> createTasks = new List<Task<ExternalResult<MetadataType>>>();
+            IList<Task<ExternalResult<MetadataType>>> updateTasks = new List<Task<ExternalResult<MetadataType>>>();
             foreach (var metadataTypeLabel in model.MetadataTypes)
```

Comment:
We should also delete metadata if the remove_lingering_metadata_types is set to true

Reply:
nvm, line 87 covers that.